packages/firestore/src/local/local_documents_view.ts

```diff
@@ -218,29 +221,38 @@ export class LocalDocumentsView {
         );
       })
       .next(matchingMutationBatches => {
-        for (const batch of matchingMutationBatches) {
-          for (const mutation of batch.mutations) {
-            const key = mutation.key;
-            // Only process documents belonging to the collection.
-            if (!query.path.isImmediateParentOf(key.path)) {
-              continue;
-            }
-
-            const baseDoc = results.get(key);
-            const mutatedDoc = mutation.applyToLocalView(
-              baseDoc,
-              baseDoc,
-              batch.localWriteTime
-            );
-            if (mutatedDoc instanceof Document) {
-              results = results.insert(key, mutatedDoc);
-            } else {
-              results = results.remove(key);
+        mutationBatches = matchingMutationBatches;
+        // It is possible that a PatchMutation can make a document a match to the query,
+        // but the version in `remoteDocumentCache` is not a match yet (waiting for server
+        // to ack). To handle this, we find all document keys affected by some`PatchMutation`s
+        // that are not in `result` yet, and back fill them via `remoteDocumentCache.getEntries`,
+        // otherwise those `PatchMutations` will be ignored because no base document can be found,
+        // and lead to missing result for the query.
+        return this.getDocumentWithPendingPatches(transaction, mutationBatches, results)
+          .next(docsWithPendingPatches => {
+            docsWithPendingPatches.forEach((key, doc) => {
+              if (doc !== null && doc instanceof Document) {
+                results = results.insert(key, doc);
+              }
+            });
```

Comment:
Nit: Add empty line to convey that the loop below does something very different (we move from one stage to another).

Reply:
Done.